### DISCUSSION TITLE
`bun test` in a monorepo

### DISCUSSION THREAD

**Discussion by vitch** [oven-sh/bun#11762] [Q&A]
I'm currently running `bun test` from the root of a bun monorepo and it's running all of my tests. 

But now I need to do some cleanup `afterEach` test in a specific sub package and I can't figure out how to do this...

I can get it to work if I run `bun test --preload` in that particular package (or put a `bunfig.toml` in that package and run locally there) but I want it to work when running all tests across the monorepo and with my `afterEach` only happening after every test _within_ this package.

I got it working pretty hackily like this:

```
// in root package.json
{
  "scripts": {
    "test": "bun run  --filter '*' test",
  }
}

// in sub-package/package.json
{
  "scripts": {
    "test": "bun test",
  }
}

// In sub-package/bunfig.toml
[test]
  root = "test"
  preload = ["./test/setup.ts"]
```

Then run `bun run test` at the root (**not** `bun test`).

But it doesn't feel ideal (not least because the output formatting is much worse).

Is there a better way? Should `bun test` in a monorepo just run `bun test` in each sub-package rather than searching for files to test from the root of the monorepo?

**Comment by Jarred-Sumner** [comment]
you could do

```sh
bun test packages/bacon packages/lettuce
```


This will match any test files that start with the path `packages/bacon` or `packages/lettuce`

edit: this won't work within package's preloads

**Reply by vitch** [reply]
I just tried but as you say in the edit, this doesn't respect the `bunfig.toml` for the nested package. 

AFAICT it is just restricting the search paths for test files. I am looking for a way to respect the fact that a monorepo is made up of separate packages which may need different configurations...

**Comment by zeyuri** [comment]
hey @Jarred-Sumner have this limitation of the preload been addressed? Its a really nice feature that is holding me up, might need to move to vitest because of this
